File: sdks/java/src/main/java/dev/atap/sdk/exception/ATAPProblemException.java
package dev.atap.sdk.exception;

import dev.atap.sdk.model.ProblemDetail;

/**
 * Exception wrapping an RFC 7807 Problem Details response.
 */
public class ATAPProblemException extends ATAPException {

    private final ProblemDetail problem;

    public ATAPProblemException(ProblemDetail problem) {
        super(problem.getDetail() != null ? problem.getDetail() : problem.getTitle(), problem.getStatus());
        this.problem = problem;
    }

    public ProblemDetail getProblem() {
        return problem;
    }

    @Override
    public String toString() {
        return "[" + problem.getStatus() + "] " + problem.getTitle() + ": "
                + (problem.getDetail() != null ? problem.getDetail() : "");
    }
}

File: sdks/php/src/Exception/ATAPProblemException.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Exception;

use Atap\Sdk\Model\ProblemDetail;

/**
 * Error wrapping an RFC 7807 Problem Details response.
 */
class ATAPProblemException extends ATAPException
{
    public function __construct(
        public readonly ProblemDetail $problem,
        ?\Throwable $previous = null,
    ) {
        parent::__construct(
            $problem->detail ?? $problem->title,
            $problem->status,
            $previous,
        );
    }

    public function __toString(): string
    {
        return sprintf('[%d] %s: %s', $this->problem->status, $this->problem->title, $this->problem->detail ?? '');
    }
}
